parlai/tasks/self_chat/worlds.py

```diff
@@ -186,3 +180,24 @@ def parley(self):
 
         self.update_counters()
         self.turn_cnt += 1
+
+    def _end_episode(self):
+        """
+        Apply logic to end the episode.
+        """
+        self.turn_cnt = 0
+        self.episode_cnt += 1
+        self.contexts = None
+        self.seed_utterances = None
+        agents = self.get_agents()
+        for a in agents:
+            a.reset()
```

Comment:
maybe we should wrap this into self.reset?

Reply:
I've created a `World.reset_agents()` method - is this what you were thinking? We can't call `World.reset()` directly here, because there's additional logic there - i.e. `World.reset()` also resets `self.total_parleys`, `self.total_exs`, `self.time`, etc.